api/buckets/service.go

```diff
@@ -1606,33 +1607,87 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 		return nil, fmt.Errorf("parsing cid path: %s", err)
 	}
 
-	ffsi, err := s.Collections.FFSInstances.Get(ctx, req.GetKey())
-	if err != nil {
-		return nil, fmt.Errorf("getting ffs instance data: %s", err)
+	var ffsInfo *mdb.FFSInfo
+	var owner crypto.PubKey
+	var isAccount bool
+	if acct := accountFromContext(ctx); acct != nil {
+		ffsInfo = acct.FFSInfo
+		owner = acct.Key
+		isAccount = true
+	} else if user := userFromContext(ctx); user != nil {
+		ffsInfo = user.FFSInfo
+		owner = user.Key
+		isAccount = false
+	}
```

Comment:
Bucket archiving now get the FFS instance via the `Account` or `User`. Building up the information we need here.

Reply:
Nit: Shouldn't happen, but maybe would be safer to add an `else` and fail fast?

Reply:
Think you probably realized this below, but we don't want to fail. In the case of no ffs info, we want to create a ffs instance.

Reply:
ah, wait. I see. the else case would be "no user found". that would be an error. will add.